server/@types/dto/index.d.ts

```diff
@@ -28,14 +30,15 @@ declare module 'dto' {
     goalReference: string
     title: string
     steps: Array<UpdateStepDto>
+    status: GoalStatusValue
```

Comment:
The core problem is that the DTO and form objects for updating a goal did not previously include the goal status, hence it could not be displayed onscreen.

The fix is to add the fields to the DTO and form object. In doing so I have also taken the opportunity to use an enum for the statuses for Goal and Step, and ensure they are populated via the mappers etc.